runtime/src/bank.rs

```diff
@@ -2001,23 +2002,31 @@ impl Bank {
         let collector_fees = self.collector_fees.load(Relaxed) as u64;
 
         if collector_fees != 0 {
-            let (unburned, burned) = self.fee_rate_governor.burn(collector_fees);
+            let (unburned, mut burned) = self.fee_rate_governor.burn(collector_fees);
             // burn a portion of fees
             debug!(
                 "distributed fee: {} (rounded from: {}, burned: {})",
                 unburned, collector_fees, burned
             );
 
-            let post_balance = self.deposit(&self.collector_id, unburned);
-            if unburned != 0 {
-                self.rewards.write().unwrap().push((
-                    self.collector_id,
-                    RewardInfo {
-                        reward_type: RewardType::Fee,
-                        lamports: unburned as i64,
-                        post_balance,
-                    },
-                ));
+            match self.deposit(&self.collector_id, unburned) {
+                Ok(post_balance) => {
+                    if unburned != 0 {
+                        self.rewards.write().unwrap().push((
+                            self.collector_id,
+                            RewardInfo {
+                                reward_type: RewardType::Fee,
+                                lamports: unburned as i64,
+                                post_balance,
+                            },
+                        ));
+                    }
+                }
+                Err(_) => {
+                    error!("Incinerated {} fee lamports", unburned);
```

Comment:
@jackcmay @t-nelson this one is special here. It follows the pattern from the other incinerator pr which we haven't closed on yet. @mvines 
I'm not entirely sure these are right and if we really need the incinerator account vs decrementing cap as the code already does in some conditions.

Reply:
Seems fine, be nice to have @mvines eyes on it since he added this

Reply:
This looks fine to me.  The `unburned` variable name is now super confusing though: `burned += unburned;` 🙃 
Consider renaming `unburned` to something like `collector_fee_after_burn`